compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -288,7 +288,13 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
 
         let infcx = self.selcx.infcx();
 
-        if obligation.predicate.has_projections() {
+        // Normalizing WellFormed predicates is not sound, see #100041
+        if obligation.predicate.has_projections()
+            && !matches!(
```

Comment:
This shouldn't be necessary once @lcnr's PR #99217 lands, and we should use `obligation.predicate.allow_normalization()` here instead.